healvis/observatory.py

```diff
@@ -230,20 +240,24 @@ def set_beam(self, beam="uniform", freq_interp_kind="linear", **kwargs):
         Set the beam of the array.
 
         Args:
-            beam : str
-                Input to PowerBeam or AnalyticBeam. If beam is
-                a viable input to AnalyticBeam, then instantiates
-                an AnalyticBeam, otherwise assumes beam is a filepath
-                to a beamfits and instantiates a PowerBeam.
+            beam : str, or list of beam objects
+                str: If it is a viable input to AnalyticBeam, 
+	            then instantiates an AnalyticBeam, otherwise assumes beam is 
+                    a filepath to a beamfits and instantiates a PowerBeam.
+                list: List of beam objects. This allows for external beams to be 
+                    used, and different beams for each antenna. They should not be 
+                    power beams. Each beam must have an interp method: 
+                    interp(self, az_array, za_array, freq_array)
             freq_interp_kind : str
                 For PowerBeam, frequency interpolation option.
 
             kwargs : keyword arguments
                 kwargs to pass to AnalyticBeam instantiation.
         """
-        if beam in ["uniform", "gaussian", "airy"] or callable(beam):
-            self.beam = AnalyticBeam(beam, **kwargs)
 
+        if isinstance(beam, list): self.beam = beam
```

Comment:
If statements should be multi-line.

It also might be good to check here that the beam object has an `interp` method.